dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -1180,30 +1080,96 @@ public Object putIfAbsent(String key, Object value) {
             }
+            return false;
         }
         return super.remove(key, value);
     }
     @Override
     public Object compute(String key, BiFunction<? super String, ? super Object, ? extends Object> remappingFunction) {
-        throw new UnsupportedOperationException();
+        Integer index = KEYMAP.getOrDefault(key, KEY_NOT_FOUND);
+        if (index != KEY_NOT_FOUND) {
+            Object newValue = remappingFunction.apply(key, propertyValues[index] == NOT_FOUND ? null : propertyValues[index]);
+            if (propertyValues[index] != NOT_FOUND) {
+                if (newValue != null) {
+                    propertyValues[index] = newValue;
+                } else {
+                    // should we set to null (since we allow null values) or NOT_FOUND (javadoc seems to suggest this)? 
+                    propertyValues[index] = NOT_FOUND;
+                }
+            } else {
+                if (newValue != null) {
+                    propertyValues[index] = newValue;
+                } else {
+                    return null;
+                }
+            }
+            return newValue;
+        }
+        return super.compute(key, remappingFunction);
     }
     @Override
     public Object computeIfAbsent(String key, Function<? super String, ? extends Object> mappingFunction) {
-        throw new UnsupportedOperationException();
+        Integer index = KEYMAP.getOrDefault(key, KEY_NOT_FOUND);
+        if (index != KEY_NOT_FOUND) {
+            if (propertyValues[index] == NOT_FOUND) {
+                Object newValue = mappingFunction.apply(key);
+                if (newValue != null) {
+                    // should we also add the newValue if null???
+                    propertyValues[index] = newValue;
+                }
+                return newValue;
+            }
+            return getFromPropertyArray(index);
+        }
+        return super.computeIfAbsent(key, mappingFunction);
     }
     @Override
     public Object computeIfPresent(String key, BiFunction<? super String, ? super Object, ? extends Object> remappingFunction) {
-        throw new UnsupportedOperationException();
+        Integer index = KEYMAP.getOrDefault(key, KEY_NOT_FOUND);
+        if (index != KEY_NOT_FOUND) {
+            if (propertyValues[index] != NOT_FOUND) {
+                Object oldValue = propertyValues[index];
+                Object newValue = remappingFunction.apply(key, oldValue);
+                if (newValue != null) {
+                    // should we also add the newValue if null, or do the remove like javadoc suggests???
+                    propertyValues[index] = newValue;
+                }
+                return newValue;
+            }
+            return null;
+        }
+        return super.computeIfPresent(key, remappingFunction);
     }
     @Override
     public Object merge(String key, Object value, BiFunction<? super Object, ? super Object, ? extends Object> remappingFunction) {
```

Comment:
If value == null or remappingFunction == null, this method needs to throw a NullPointerException.  Should check that first in the method.

The if (propertyValues[index] != NULL) should be if (propertyValues[index] != NULL || propertyValues[index] == null)

if newValue == null it should be NOT_FOUND, i.e. removed.  So similarly to above it should be

propertyValues[index] = newValue == null ? NOT_FOUND : newValue;

return propertyValues[index]  should be changed to return getFromPropertyArray(index);